dbms/src/Common/Pipe.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <fcntl.h>
+#include <dlfcn.h>
+#include <Common/Exception.h>
+#include <Common/ShellCommand.h>
+#include <Poco/Logger.h>
+#include <common/logger_useful.h>
+#include <IO/WriteHelpers.h>
+#include <port/unistd.h>
+#include <csignal>
+
+namespace DB {
+    namespace ErrorCodes
+    {
+        extern const int CANNOT_PIPE;
+    }
+
+    struct Pipe
+    {
+        union
+        {
+            int fds[2];
+            struct
```

Comment:
This is not compliant with C++ standard (nested anonymous struct/union; you'll see warnings from clang).